### ISSUE TITLE
U(1)-Sz single-site CTM collapses env to exact-zero (E=0) at D=3 χ=12 — regression from PR #671 sorted-tail tiling

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#700]
> 🤖 **AI-generated issue** — written by Claude Code, posted by @yingjerkao.

## Summary

`ctm_tensor` (single-site U(1)-Sz path) silently returns an **exact-zero environment** — and therefore `E = 0` — for a **healthy, non-degenerate random init** at **D=3, χ=12**. Bisected to a **clean regression** in PR #671 (`004fc89`, `fix(#667): canonical env tiling + 2x2 init corner consistency`).

This is distinct from #666 (that one collapses on *near-degenerate SU-converged* states at sweep 3; this one collapses on a *healthy random* init at sweep 1 and is a git-bisected tiling regression).

## Reproduce

```python
import jax, numpy as np
jax.config.update("jax_enable_x64", True)
from tenax.algorithms._ctm_tensor import ctm_tensor
from tenax.algorithms.ipeps import heisenberg_u1sz_init_pair
A, _ = heisenberg_u1sz_init_pair(D=3, key=jax.random.PRNGKey(0))
env, _ = ctm_tensor(A, chi=12, max_iter=5, conv_tol=1e-12)
print(float(np.linalg.norm(np.asarray(env.C1._data))))   # -> 0.0  (should be ~1.0)
```

The init env is healthy (`C1` norm 1.0, correct block counts). The env is zeroed on the **first** absorption sweep.

## Bisect + mechanism

`git bisect` (good `07e60c5`, bad `main`) → first bad commit **`004fc89` (PR #671)**.

That commit replaced enumeration-order chi-leg padding with **sorted-tail** padding (`_tile_fused_to_chi` in `src/tenax/algorithms/_ctm_tensor_init.py`): it keeps the leading D² block in enumeration order but appends the `χ − D²` padding charges in **sorted** order. For a direction-uniform single-site tensor with χ > D² this changes which charges occupy the padding slots, and for **D=3 χ=12** it drives the first-sweep projector/absorption to zero the environment.

The trigger is **padding-specific**, not simply "χ > D²":

| D | D² | χ | pad | result |
|---|----|----|-----|--------|
| 3 | 9  | 10 | 1 | ✓ nonzero |
| 3 | 9  | **12** | **3** | ✗ **collapse (E=0)** |
| 4 | 16 | 18 | 2 | ✓ nonzero |

Energy before/after the regression on the reproducer (`heisenberg_gate_u1sz`, χ=12, max_iter=20):
- `07e60c5` (pre-#671): **E = −0.0617** (sane, non-zero env)
- current `main`: **E = 0.0** (zero env)

## Notes on the fix

- PR #671's own direction-dependent acceptance test is **xfailed** (deferred to #670), so the sorted-tail tiling that caused this collapse is not yet delivering working direction-dependent CTM — i.e. the change currently costs a single-site regression with no realized benefit.
- Any real fix must preserve #671's Phase 0/1 corner↔edge charge-multiset consistency (the stated goal of the sorted tiling) while not collapsing the direction-uniform χ > D² single-site env. Candidate directions: restore enumeration-order padding for the direction-uniform case, or make corner/edge padding jointly consistent without perturbing the projector's kept sectors.

## Impact

Silent (`E = 0`, no error raised) for any single-site U(1)-Sz `ctm_tensor` that lands on the affected (D, χ) padding regime. Discovered via the throwaway `test_u1sz_defrag_prototype_610` guards (#692), which are being xfailed with a pointer here pending this fix.

Related: #666 (zero-corner on near-degenerate SU states), #667/#670/#671 (direction-dependent symmetric CTM).
